redash/tasks/queries.py

```diff
@@ -441,9 +457,13 @@ def run(self):
 
     def _annotate_query(self, query_runner):
         if query_runner.annotate_query():
+            if self.task.request.delivery_info:
```

Comment:
:\ I don't like changing code to accommodate for tests (this will never be executed outside of a celery context otherwise). Two alternatives:

- Pass the routing key & task id to the constructor instead of the task directly. 
- Mock this.

WDYT?